ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -145,28 +170,89 @@ export const ReviewPermissions = () => {
     setShowNetworkToast(true);
   };
 
-  const accounts = useSelector(getUpdatedAndSortedAccounts);
-  const evmAccounts: MergedInternalAccount[] = useMemo(() => {
-    return accounts.filter((account: InternalAccount) =>
-      isEvmAccountType(account.type),
-    );
-  }, [accounts]);
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccounts,
+  ) as MergedInternalAccount[];
+
+  const allAccountsWithCaipAccountId = allAccounts.map((account) => {
+    // I hope we can reliably use the first scope to determine the namespace
+    const { namespace, reference } = parseCaipChainId(account.scopes[0]);
+    return {
+      internalAccount: account,
+      caipAccountId:
+        `${namespace}:${reference}:${account.address}` as CaipAccountId,
+    };
+  });
 
-  const connectedAccountAddresses = useSelector((state) =>
-    getPermittedAccountsForSelectedTab(state, activeTabOrigin),
-  ) as string[];
+  const _connectedAccountAddresses = useSelector((state) =>
+    getAllPermittedAccountsForSelectedTab(state, activeTabOrigin),
+  ) as CaipAccountId[];
 
-  const handleSelectAccountAddresses = (addresses: string[]) => {
-    if (addresses.length === 0) {
+  // should this be at the selector level? :|
+  const connectedAccountAddresses = uniq(
+    _connectedAccountAddresses.map((caipAccountId) => {
+      const {
+        address,
+        chain: { namespace, reference },
+      } = parseCaipAccountId(caipAccountId);
+      if (namespace === KnownCaipNamespace.Eip155) {
+        // this is very hacky, but it works for now
+        return `${namespace}:0:${address}` as CaipAccountId;
+      }
+      return `${namespace}:${reference}:${address}` as CaipAccountId;
+    }),
+  );
```

Comment:
I think massaging this this way is fine for now. but should we not be doing similar for Solana?..

Here, an EVM address assigned to multiple scopes gets uniqed into `eip155:0:address`, but the non evm addresses assigned to multiple scopes do not get uniqed and one will exist for each scope

Reply:
Looking at this